crates/exex/exex/src/wal/cache.rs

```diff
@@ -8,14 +11,14 @@ use reth_primitives::{BlockNumHash, B256};
 ///
 /// This cache is needed to avoid walking the WAL directory every time we want to find a
 /// notification corresponding to a block or a block corresponding to a hash.
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 pub struct BlockCache {
```

Comment:
this type now has two locked fields.
arent those two fields in sync, meaning `files` and `blocks` have the same entries?
not obvious from reading the docs.

Reply:
previously we've deleted the notifications from WAL by walking them in an order they were inserted, so we needed a `BTreeMap` indexed by the file ID

we won't need it anymore and the `files` will be removed https://github.com/paradigmxyz/reth/pull/11266/files#r1777654333

basically, having a mapping of blocks (either hashes to file IDs, or numbers to file IDs) is enough for our new design https://github.com/paradigmxyz/reth/issues/11202 because we don't rely on the order of files anymore and walk by hashes